go/vt/mysqlctl/s3backupstorage/s3.go

```diff
@@ -496,6 +497,19 @@ func objName(parts ...string) *string {
 	return &res
 }
 
+// This creates a new transport based off http.DefaultTransport the first time and returns the same
+// transport on subsequent calls so connections can be reused as part of the same transport.
+func getS3Transport() *http.Transport {
+	if defaultS3Transport == nil {
+		tlsClientConf := &tls.Config{InsecureSkipVerify: tlsSkipVerifyCert}
+
+		defaultS3Transport = http.DefaultTransport.(*http.Transport).Clone()
+		defaultS3Transport.TLSClientConfig = tlsClientConf
+	}
+
+	return defaultS3Transport
```

Comment:
Is this concurrency safe / does it need to be? Since it doesn't seem like it is. 

I'm also not a big fan of having a global for this, those tend to bite us later on. Can we move it at least into `S3BackupStorage` instead then? That's still kinda global, but at least a tad better then.

Reply:
We can then also initialize it already in the `init()` to avoid the whole concurrency question / potential problem?

Reply:
yes, it should be, specially considering the default on the SDK already uses `http.DefaultClient`, which in turn uses `http.DefaultTransport`, this would have been a shared default in most projects that don't define their transport/client from scratch, see the `Config` [docs](https://docs.aws.amazon.com/sdk-for-go/api/aws/#Config).

I thought the `S3BackupStorage` would be created for each request being processed, but if that is not the case we could move it there. I will test it out and report back.

if the above is not possible and setting it up on `init()` is preferred I can do that too

Reply:
I see that you added a unit test for this. What's the conclusion?
@dbussink time for a re-review?

Reply:
@deepthi I’m back next week but I don’t see a refactor of this or a reason why that couldn’t be done. I strongly prefer not having a global like this if that can reasonably be avoided. 

Also happy to push up that refactor once I’m back.

Reply:
I have moved this to inside the `S3BackupStorage` and re-run our tests and seems the connections are getting re-used, so I have update the PR to reflect that, thanks for pointing it out!